src/components/Table/utils/RowState.ts

```diff
@@ -234,6 +307,28 @@ export class RowState {
     return typeof this.row.pin === "string" || (!!this.row.pin && this.row.pin.filter !== true);
   }
 
+  // mobx will cache this getter for us
+  public get api(): GridRowApi<R> {
+    const rs = this;
+    // Copy the GridTableApi + the getVisibleChildren GridRowApi method
+    return {
+      ...this.states.table.api,
+      // The caller can invoke this observable without their own useComputed,
+      // b/c we wrap all rows in an observer
+      getVisibleChildren(kind?: R["kind"]): GridDataRow<R>[] {
+        // Avoid infinite loop if a cell asks for getVisibleChildren while calculating isMatched
+        const children = rs.isCalculatingDirectMatch
+          ? rs.visibleDirectlyMatchedChildren.map((cs) => cs.row)
+          : rs.visibleChildren.map((cs) => cs.row);
+        return !kind ? children : children.filter((r) => r.kind === kind);
+      },
+      getSelectedChildren(kind?: R["kind"]): GridDataRow<R>[] {
+        const children = (rs.children ?? []).filter((cs) => cs.isSelected).map((cs) => cs.row);
+        return !kind ? children : children.filter((r) => r.kind === kind);
+      },
+    } as any;
```

Comment:
**question:** Why do we need this `as any`?

Reply:
Gotcha. Not a big deal. I think we do that in the TableApi file as well somewhere.